compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/JvmBuiltinOptimizationLowering.kt

```diff
@@ -120,6 +120,26 @@ class JvmBuiltinOptimizationLowering(val context: JvmBackendContext) : FileLower
                         putValueArgument(0, expression.branches[0].result)
                     }
                 }
+                if (expression.origin == IrStatementOrigin.OROR) {
+                    assert(
+                        expression.type.isBoolean()
+                                && expression.branches.size == 2
+                                && expression.branches[0].result.isTrueConst()
+                                && expression.branches[1].condition.isTrueConst()) {
+                        "OROR condition should have an 'if a then true' body on its first branch, " +
+                                "and an 'if true then b' body on its second branch. " +
+                                "Failing expression: ${expression.dump()}"
+                    }
+                    return IrCallImpl(
+                        expression.startOffset,
+                        expression.endOffset,
+                        context.irBuiltIns.booleanType,
+                        context.irIntrinsics.ororSymbol
+                    ).apply {
+                        dispatchReceiver = expression.branches[0].condition
+                        putValueArgument(0, expression.branches[1].result)
+                    }
```

Comment:
Although this call doesn't exist in bytecode, is there any reason preferring dispatchReceiver + arg0 over arg0 + arg1? That seems to be inconsistent with the parameters passed to `Key()`.

Reply:
It was intended to make it read like a.and(b), but I defined its symbol in the format of 2 value arguments, so should be changed to align with it. Was not captured before as Intrinsic does not go through expressionCodegen. I will change it to be 2 value arguments here, for andand as well.